File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/reservations/kafka/ReservationCommandProducer.java
package de.vkb.dojo.es.reservations.kafka;

import de.vkb.dojo.es.reservations.kafka.config.TopicNames;
import de.vkb.dojo.es.reservations.model.command.ReservationCommand;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

@Component
public class ReservationCommandProducer {
    @Autowired
    TopicNames topicNames;
    @Autowired
    KafkaTemplate<String, ReservationCommand> template;

    public void publish(ReservationCommand command) {
        template.send(topicNames.getReservationCommand(), command.getOperationId(), command);
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/reservations/kafka/ReservationCommandProducer.kt
package de.vkb.dojo.es.reservations.kafka

import de.vkb.dojo.es.reservations.model.command.ReservationCommand
import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.configuration.kafka.annotation.KafkaKey
import io.micronaut.configuration.kafka.annotation.Topic

@KafkaClient(
    id = "\${kafka.producers.reservation-command.id}",
    acks = KafkaClient.Acknowledge.ALL
)
interface ReservationCommandProducer {
    @Topic("\${kafka.topics.reservationCommand}")
    fun sendCommand(@KafkaKey key: String, command: ReservationCommand)
}
